csharp/extractor/Semmle.Extraction.CSharp/Entities/Method.cs

```diff
@@ -336,17 +342,21 @@ protected void ExtractGenerics()
                 {
                     Context.Emit(Tuples.is_constructed(this));
                     Context.Emit(Tuples.constructed_generic(this, Method.Create(Context, ConstructedFromSymbol)));
-                    foreach (var tp in symbol.TypeArguments)
+                    foreach (var tp in symbol.GetAnnotatedTypeArguments())
                     {
-                        Context.Emit(Tuples.type_arguments(Type.Create(Context, tp), child++, this));
+                        Context.Emit(Tuples.type_arguments(Type.Create(Context, tp.Symbol), child, this));
+                        if(tp.Nullability.NeedsExtraction())
+                            Context.Emit(Tuples.type_argument_annotation(this, child, (Kinds.TypeAnnotation)symbol.TypeArgumentsNullableAnnotations[child]));
```

Comment:
Why not `tp.Nullability` instead of `symbol.TypeArgumentsNullableAnnotations[child]`?